### PR TITLE
fix(lsp): `textDocument/references` should respect `includeDeclaration`

### PR DESCRIPTION
**Pull request by dsherret**
Noticed this while I was writing tests in a separate PR.

### CODE REVIEW

File: cli/tests/integration/lsp_tests.rs [hunk 1]
```diff
@@ -2428,18 +2428,12 @@ fn lsp_code_lens() {
...
         "arguments": [
           "file:///a/file.ts",
           { "line": 0, "character": 6 },
           [{
-            "uri": "file:///a/file.ts",
-            "range": {
-              "start": { "line": 0, "character": 6 },
-              "end": { "line": 0, "character": 7 }
-            }
-          }, {
```
Comment (dsherret) [inline cli/tests/integration/lsp_tests.rs:2442]: Our code lens was displaying as always having at least 1 reference, but it should be 0 when there are 0 references.

Comment (bartlomieju) [reply cli/tests/integration/lsp_tests.rs:2442]: I don't see a test for 0 reference, could you add it?

Comment (dsherret) [reply cli/tests/integration/lsp_tests.rs:2442]: This is essentially the test for that (the declaration is no longer included).

Comment (dsherret) [reply cli/tests/integration/lsp_tests.rs:2442]: Oh wait, there's a bug in the output for 0 references lol. Let me write a unit test.

Comment (dsherret) [reply cli/tests/integration/lsp_tests.rs:2442]: Oh no, it's there. I misread the code, but let me improve it so it's easier to read.

**Pull request merged**